hal/src/nRF52840/spi_hal.cpp

```diff
@@ -522,19 +523,17 @@ int32_t hal_spi_set_settings(hal_spi_interface_t spi, uint8_t set_default, uint8
 int hal_spi_sleep(hal_spi_interface_t spi, bool sleep, void* reserved) {
     if (sleep) {
         CHECK_TRUE(hal_spi_is_enabled(spi), SYSTEM_ERROR_NONE);
-        CHECK_FALSE(spiMap[spi].suspended, SYSTEM_ERROR_NONE);
         hal_spi_transfer_dma_cancel(spi);
         while (spiMap[spi].transmitting);
-        hal_spi_end(spi); // It doesn't clear spi settings, so we can reuse the previous settings on woken up.
-        // hal_spi_end() clears pin function, retrieve it.
+        spiUninit(spi);
+        // spiUninit() clears pin function, retrieve it.
         HAL_Set_Pin_Function(spiMap[spi].sck_pin, PF_SPI);
         HAL_Set_Pin_Function(spiMap[spi].mosi_pin, PF_SPI);
         HAL_Set_Pin_Function(spiMap[spi].miso_pin, PF_SPI);
-        spiMap[spi].suspended = true;
+        spiMap[spi].state = HAL_SPI_STATE_SUSPENDED;
     } else {
-        CHECK_TRUE(spiMap[spi].suspended, SYSTEM_ERROR_NONE);
+        CHECK_TRUE(spiMap[spi].state == HAL_SPI_STATE_SUSPENDED, SYSTEM_ERROR_NONE);
```

Comment:
Let's make sure we have the same behavior across all APIs. Sometimes we return INVALID_STATE, sometimes it's ERROR_NONE.